src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1467,6 +1489,11 @@ MasterAggregateExpression(Aggref *originalAggregate,
 		unionAggregate->aggtype = hllType;
 		unionAggregate->args = list_make1(hllTargetEntry);
 		unionAggregate->aggkind = AGGKIND_NORMAL;
+#if (PG_VERSION_NUM >= 90600)
+		unionAggregate->aggtranstype = InvalidOid;
+		unionAggregate->aggargtypes = list_make1_oid(unionAggregate->aggtype);
+		unionAggregate->aggsplit = AGGSPLIT_SIMPLE;
```

Comment:
Possibly not now, but it seems we should combine the code for all these cases to some degree.